go.mod

```diff
@@ -1,14 +1,20 @@
 module github.com/gravitational/trace
 
-go 1.15
+go 1.20
```

Comment:
Dependencies want various increased Go versions, as follows:

* testify wants 1.20 - https://github.com/stretchr/testify/blob/v1.8.3/go.mod
* /x/net wants 1.18 (updated by this PR) - https://cs.opensource.google/go/x/net/+/refs/tags/v0.33.0:go.mod
* grpc wants 1.17 - https://github.com/grpc/grpc-go/blob/v1.56.3/go.mod

These are all fairly old, past EOL Go versions. Since this is a library it makes sense to keep Go requirements as soft as can be, so I've updated trace to require 1.20. This and the security patch probably both warrant a new tag.